filebeat/input/filestream/internal/input-logfile/harvester.go

```diff
@@ -60,11 +64,20 @@ type defaultHarvesterGroup struct {
 // Run starts the Harvester for a Source.
 func (hg *defaultHarvesterGroup) Run(ctx input.Context, s Source) error {
 	log := ctx.Logger.With("source", s.Name())
-	log.Debug("Starting harvester for file")
 
 	harvesterCtx, cancelHarvester := context.WithCancel(ctxtool.FromCanceller(ctx.Cancelation))
 	ctx.Cancelation = harvesterCtx
 
+	// this blocks Run until a new reader can be started for the file
+	limitErr := hg.limitSem.Acquire(harvesterCtx, 1)
+	if limitErr != nil {
+		log.Debug("Failed to start harvester as limit has been reached.")
```

Comment:
You got me confused here? The semaphore should block until enough resources become available right? An error != nil here indicates that the context was closed.